### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by PatrykSzwer** [StarcounterApps/Website#118]
This represents the Roadmap for Website app changes, from bug fixes to visual changes.

### Roadmap

1. [x] 1. Refactoring and initial changes
    1. [x] 1. ~~Rename the app "Website" to "WebsiteEditor #34~~
    2. [x] 2. ~~Rename "*Template" to "*Surface" #114~~
<p>

2. [x] 2. Investigation and fixing of discovered bugs
    1. [x] 1. ~~Save button does not get enabled for a newly signed in user. #102~~
    2. [x] 2. ~~Pinning rule is not unmapped correctly #101~~
    3. [x] 3. ~~Signup icon is not visible with Latest SC RC (2.3.1.6410) #95 (Need confirmation)~~
    4. [x] 4. ~~Cannot change parent of objects in Typed JSON trees #88 (Need further explanation)~~
    5. [x] 5. ~~Website Provider is detected by Avast antivirus as threat #76~~
    6. [x] 6. ~~NullReferenceException when Pinning Rule refers to a static html #56~~
    7. [x] 7. ~~Moving to a page with not final Catching rule reloads whole Surface #55~~
    8. [x] 8. ~~Remapped URLs should appear only once #14 (Out of scope)~~
<p>

3. [x] 3. ~~Release of new, bug-free version and deployment at inside.starcounter.com~~
4. [x] 4. ~~2.3.0 Starcounter release support - Fix on IsHandlerRegistered usage #108 (Done, waiting for merge)~~
<p>

5. [x] 5. Visual changes and enhancements
    1. [x] 1. ~~Discuss and design new UI for Website #109~~
    2. [x] 2. ~~Implement new UI navigation and separate views based on one surface scope concept #109~~
    3. [x] ~~3. Merge Pinning rules view with blending points as accordion/collapse rows. #109~~
    4. [x] ~~4.  Support groups of surfaces #122~~
<p>

6. [x] ~~6. Release of new, visually changed version and deployment at inside.starcounter.com~~
7. [x] ~~7. Allow more catching rules than just URI #48~~
8. [x] ~~8. Release of new version with more catching rules feature and deployment at inside.starcounter.com.~~
9. [x] ~~9. Add possibility to move surfaces between groups.~~
10. [x] ~~10. Deleted pinning rules get unmapped before commiting transaction #140~~
11. [x] ~~11. Error when saving a pinning rule with same properties #139~~
12. [x] ~~12. Change namespaces to be compliant with folder structure #130~~
13. [x] ~~13. Selenium tests for adding, deleting and changing pinning rules. #126~~
<p>

14. [x] 14. Enhancements
    1. [x] ~~1. Implement Authorization library #123~~
    2. [x] ~~2. Implement new UI for catching rules #150~~
    3. [x] ~~3. Release new version and deploy it at inside.starcounter.com~~
    4. [x] ~~4. Allow disabling of stuff #116~~
    5. [x] ~~5. Add predefined configs for surfaces #113~~
    6. [x] ~~6. Add an example of a responsive surface #131~~
<p>

15. [x] ~~15. Release of new, improved version and deployment at inside.starcounter.com.~~
<p>

16. [ ] 16. Functional enhancements
    1. [x] ~~1. Use GetHandlerForUri instead of current implementation of GetWildCardUrl #94~~
    2. [ ] 2. Add Workspaces to the Website app #47
    3. [x] ~~3. Support regex in Catch Uris #54~~
    4. [x] ~~4. It is not possible to blend main response and pinned content #97 (LP)~~
    5. [x] ~~5. Question : Is there a way to avoid page flicker on clicking/changing tabs? #62~~
<p>

17. [ ] 17. Release of final, enhancement version and deployment at inside.starcounter.com.

**Comment by warpech** [comment]
I think that #108 breaks compat with Starcounter 2.3.0. In my opinion this can be done after "3. Release of new, bug-free version and deployment at inside.starcounter.com"

**Comment by PatrykSzwer** [comment]
Ok, sounds legit. I've changed the order.

**Comment by warpech** [comment]
I have added priority labels to the issues. I think that #97 should not be worked on right now, because it is a lot of effort and little benefit right now. Pls move it to "6. Functional enhancements"

On the other hand, this should be prioritised: #48. Pls move it as the first thing in "6. Functional enhancements" or even make a special release directly after doing #48.

**Comment by PatrykSzwer** [comment]
Great, I wanted to prioritize them too, but I wasn't sure about specific issues priorities. Thanks for that.

**Comment by warpech** [comment]
#48 and #122 are needed before July 17th.

**Comment by PatrykSzwer** [comment]
As we discussed on last meeting with @warpech and @un-tone I've created new, test milestone with current issues and features that We want to be done for the next release. 
Milestone can be found here: https://github.com/StarcounterApps/Website/milestone/1

**Comment by PatrykSzwer** [comment]
Created new milestone based on current issues: https://github.com/StarcounterApps/Website/milestone/2

**Comment by warpech** [comment]
That was a good initiative with the Milestones, but we need to coordinate even more accross the repos. From now on, our Roadmap for [all the apps of the Web Platform Team](https://github.com/search?q=topic%3Aweb-platform-team) will be mainained in a project: https://github.com/orgs/StarcounterApps/projects/3
